### DISCUSSION TITLE
Write Defaults

### DISCUSSION THREAD

**Discussion by DasLixou** [charleskorn/kaml#281] [Q&A] [answered]
Hello!
I use Kaml and wonder if i can use something like `Yaml.default.writeDefaults(obj)` so that it only overrides things that are missing or have the wrong type (e.g. String != Int). Or is it possible to dont throw errors when reading and just use the default values? 

Thanks :D

**Comment by charleskorn** [comment]
> I use Kaml and wonder if i can use something like Yaml.default.writeDefaults(obj) so that it only overrides things that are missing or have the wrong type (e.g. String != Int).

I'm not quite sure what you're after sorry - could you perhaps illustrate your question with an example to help me understand?

> Or is it possible to dont throw errors when reading and just use the default values?

This behaviour would have to be supported by kotlinx.serialization - I'd suggest raising this as a feature request on its [repo](https://github.com/Kotlin/kotlinx.serialization).

**Reply by DasLixou** [reply]
Yeah so i just want to read some configs from my config.yml file, but when i delete e.g. the port key-value, i dont want to throw an error and just use the default value from my Config Object

**Reply by charleskorn** [reply]
If you'd like a property to have a default value, you should be able to just set the default value on the property - see https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/basic-serialization.md#optional-properties.

**Reply by DasLixou** [reply]
Weird, I already have a default value but still it says its required. Hmm I’ll update dependencies and search bit more

**Reply by DasLixou** [reply]
This is my current file: https://github.com/TeamSkyBeach/Stracciatella/blob/main/src/main/kotlin/cc/lixou/stracciatella/ServerConfig.kt

**Reply by DasLixou** [reply]
:o yeah its another problem because the yaml is empty.

error:
```
Exception in thread "main" com.charleskorn.kaml.EmptyYamlDocumentException at <root> on line 1, column 1: The YAML document is empty.
	at com.charleskorn.kaml.YamlParser.<init>(YamlParser.kt:41)
	at com.charleskorn.kaml.Yaml.decodeFromReader(Yaml.kt:52)
	at com.charleskorn.kaml.Yaml.decodeFromString(Yaml.kt:44)
	at cc.lixou.stracciatella.Stracciatella.<init>(Stracciatella.kt:60)
	at MainKt.main(Main.kt:4)
```

**Reply by charleskorn** [reply]
(for anyone coming across this thread, the discussion continues in #284)

**Comment by DasLixou** [comment, marked as answer]
Will be fixed in a PR with config. See #284
